Brokerages/Oanda/OandaBrokerage.cs

```diff
@@ -0,0 +1,1052 @@
+                        break;
+                }
+
+                //3 months is the max expiry for Oanda, and OrderDuration.GTC is only currently available
+                requestParams.Add("expiry", XmlConvert.ToString(DateTime.Now.AddMonths(3), XmlDateTimeSerializationMode.Utc));
+            }
+
+            //this type should contain a stop and a limit to that stop.
+            if (order.Type == OrderType.StopLimit)
+            {
+                requestParams.Add("type", "stop");
+                requestParams.Add("price", ((StopLimitOrder)order).StopPrice.ToString(CultureInfo.InvariantCulture));
+                switch (order.Direction)
+                {
+                    case OrderDirection.Buy:
+                        //Orders can be submitted with lower and upper bounds. If the market price on execution falls outside these bounds, it is considered a "Bounds Violation" and the order is cancelled.
+                        break;
+                    case OrderDirection.Sell:
+                        //Orders can be submitted with lower and upper bounds. If the market price on execution falls outside these bounds, it is considered a "Bounds Violation" and the order is cancelled.
+                        break;
+                }
+
+                //3 months is the max expiry for Oanda, and OrderDuration.GTC is only currently available
+                requestParams.Add("expiry", XmlConvert.ToString(DateTime.Now.AddMonths(3), XmlDateTimeSerializationMode.Utc));
+            }
+
+            if (order.Type == OrderType.StopMarket)
+            {
+                requestParams.Add("type", "marketIfTouched");
+                requestParams.Add("price", ((StopMarketOrder)order).StopPrice.ToString(CultureInfo.InvariantCulture));
+                switch (order.Direction)
+                {
+                    case OrderDirection.Buy:
+                        requestParams.Add("upperBound", ((StopMarketOrder)order).StopPrice.ToString(CultureInfo.InvariantCulture));
+                        break;
+                    case OrderDirection.Sell:
+                        requestParams.Add("lowerBound", ((StopMarketOrder)order).StopPrice.ToString(CultureInfo.InvariantCulture));
+                        break;
+                }
+
+                //3 months is the max expiry for Oanda, and OrderDuration.GTC is only currently available
+                requestParams.Add("expiry", XmlConvert.ToString(DateTime.Now.AddMonths(3), XmlDateTimeSerializationMode.Utc));
+            }
+        }
+
+
+        /// <summary>
+        /// Checks for fill events by registering to the event session to receive events.
+        /// </summary>
+        private void CheckForFills()
+        {
+            // reentrance gaurd
+            if (!Monitor.TryEnter(_fillLock))
```

Comment:
I don't think we need the `_fillLock` here, it was used in the tradier implementation because we had to poll their api. Oanda provides even streaming. I would recommend making the EventsSession instance a class wide variable to codify it's lifetime, as it's not immediately obvious that the object will live after this method due to the event handler binding.